pkg/cmd/template/schema_test.go

```diff
@@ -711,57 +673,25 @@ rendered: true`
 	}
 }
 
-func TestNoSchemaFileSchemaFlagSet(t *testing.T) {
-	dataValuesYAML := `#@data/values
----
-db_conn:
-  hostname: server.example.com
-  port: 5432
-  username: sa
-  password: changeme
-`
-	templateYAML := `---
-rendered: true`
-
-	filesToProcess := files.NewSortedFiles([]*files.File{
-		files.MustNewFileFromSource(files.NewBytesSource("dataValues.yml", []byte(dataValuesYAML))),
-		files.MustNewFileFromSource(files.NewBytesSource("template.yml", []byte(templateYAML))),
-	})
-
-	ui := cmdcore.NewPlainUI(false)
-	opts := cmdtpl.NewOptions()
-	opts.SchemaEnabled = true
-	out := opts.RunWithFiles(cmdtpl.TemplateInput{Files: filesToProcess}, ui)
-	if out.Err == nil {
-		t.Fatalf("Expected RunWithFiles to fail with message about schema enabled but no schema provided, but was a success")
-	}
-
-	if !strings.Contains(out.Err.Error(), "Schema experiment flag was enabled but no schema document was provided") {
-		t.Fatalf("Expected an error about schema enabled but no schema provided, but got: %v", out.Err.Error())
-	}
-}
-
 func testSchemaTemplates(t *testing.T, schemaYAML string, dataValuesYAML string, templateYAML string, expected string) {
```

Comment:
Good points.
filesToProcess is now a parameter that is passed to the helper function.

Also, instead of passing an error argument as suggested, we went with splitting the 'assertion' helper function into 2 separate functions:
1) assertSchemaTemplatesSucceed
2) assertSchemaTemplatesFail